docs/en/custom-settings.md
# Custom settings

**[Русская версия здесь.]**

A list of all the parameters that can be specified when creating a utility instance.

* [horizontalPosition](#horizontalPosition)
* [verticalPosition](#verticalPosition)
* [maxWidth](#maxWidth)
* [maxHeight](#maxHeight)
* [fontSize](#fontSize)
* [showClearTitle](#showClearTitle)
* [showOutputDirection](#showOutputDirection)
* [allowColorization](#allowColorization)
* [consoleColorScheme](#consoleColorScheme)
* [popupColorScheme](#popupColorScheme)
* [serverColorScheme](#serverColorScheme)
* [consoleEOL](#consoleEOL)
* [tabChar](#tabChar)
* [decimalPlaces](#decimalPlaces)
* [run](#run)

Parameters must be passed through the object, for example:
```javascript
const UduJS = require('udujs/Client');
const Debug = new UduJS({
  maxWidth: 'auto',
  fontSize: '1.2em',
  decimalPlaces: 4,
  popupColorScheme: 'dark',
});
```

<a name="horizontalPosition"></a>
## horizontalPosition <code>string | client only</code>

Horizontal position of pop-up message on the browser window.
Valid values are <code>left</code> or <code>right</code>.
The default value is <code>left</code>.
```javascript
const Debug = new UduJS({
  horizontalPosition: 'right',
});
```

<a name="verticalPosition"></a>
## verticalPosition <code>string | client only</code>

Vertical position of pop-up message on the browser window.
Valid values are <code>top</code> or <code>bottom</code>.
The default value is <code>bottom</code>.
```javascript
const Debug = new UduJS({
  verticalPosition: 'top',
});
```

<a name="maxWidth"></a>
## maxWidth <code>number, string | client only</code>

The maximum width of the pop-up message.
Valid values are <code>number</code> or <code>auto</code>.
The default value is <code>500</code>.
```javascript
const Debug = new UduJS({
  maxWidth: 'auto',
});
```

<a name="maxHeight"></a>
## maxHeight <code>number | client only</code>

The maximum height of the pop-up message.
Valid value are <code>number</code>.
The default value is <code>600</code>.
```javascript
const Debug = new UduJS({
  maxHeight: 400,
});
```

<a name="fontSize"></a>
## fontSize <code>string | client only</code>

The font size in the pop-up message.
Valid value are <code>string</code>.
The default value is <code>1em</code>.
```javascript
const Debug = new UduJS({
  fontSize: '2em',
});
```

<a name="showClearTitle"></a>
## showClearTitle <code>boolean | client only</code>

The "Click to clear" tooltip on the pop-up message. If annoying, then you can turn it off.
Valid values are <code>true</code> or <code>false</code>.
The default value is <code>true</code>.
```javascript
const Debug = new UduJS({
  showClearTitle: false,
});
```

<a name="showOutputDirection"></a>
## showOutputDirection <code>string | client only</code>

Output direction for the <code>[show()](./client-api.md#show)</code> method.
Valid values are <code>window</code> or <code>console</code>.
The default value is <code>window</code>.
```javascript
const Debug = new UduJS({
  showOutputDirection: 'console',
});
```

<a name="allowColorization"></a>
## allowColorization <code>boolean | client only</code>

Allow coloring of the displayed information.
This parameter is applicable **only to the console on the browser**.
Turn it off if the browser does not correctly color the text in the console.
Valid values are <code>true</code> or <code>false</code>.
The default value is <code>true</code>.
```javascript
const Debug = new UduJS({
  allowColorization: false,
});
```

<a name="consoleColorScheme"></a>
## consoleColorScheme <code>string | client only</code>

The [color scheme] for the console in the browser.
Valid values are <code>dark</code>, <code>bright</code> or <code>custom</code>.
The default value is <code>dark</code>.
```javascript
const Debug = new UduJS({
  consoleColorScheme: 'bright',
});
```

<a name="popupColorScheme"></a>
## popupColorScheme <code>string | client only</code>

The [color scheme] for a pop-up message in the browser.
Valid values are <code>dark</code>, <code>bright</code> or <code>custom</code>.
The default value is <code>bright</code>.
```javascript
const Debug = new UduJS({
  popupColorScheme: 'dark',
});
```

<a name="serverColorScheme"></a>
## serverColorScheme <code>string | server only</code>

The [color scheme] for the console on the server.
Valid values are <code>dark</code>, <code>bright</code> or <code>custom</code>.
The default value is <code>dark</code>.
```javascript
const Debug = new UduJS({
  serverColorScheme: 'custom',
});
```

<a name="consoleEOL"></a>
## consoleEOL <code>string</code>

A newline character for messages in the console.
Recommended values: <code>\r\n</code> - Windows-style, <code>\n</code> - Linux-style.
Valid value are <code>string</code>.
The default value is <code>\n</code>.
```javascript
const Debug = new UduJS({
  consoleEOL: '\r\n',
});
```

<a name="tabChar"></a>
## tabChar <code>string</code>

The size of one tab character for formatting messages.
Step of indent for nested data structures.
Valid value are <code>string</code>.
The default is two space characters.
```javascript
const Debug = new UduJS({
  tabChar: '..',
});
```

<a name="decimalPlaces"></a>
## decimalPlaces <code>number</code>

The number of decimal places for displaying the execution time in RTT-methods (accuracy of measurement).
Specify 0 to disable the decimal part.
Valid value are <code>number</code>.
The default value is <code>2</code>.
```javascript
const Debug = new UduJS({
  decimalPlaces: 4,
});
```

<a name="run"></a>
## run <code>boolean</code>

This parameter allows you to quickly disable all components of the utility.
Valid values are <code>true</code> or <code>false</code>.
The default value is <code>true</code>.
```javascript
const Debug = new UduJS({
  run: false,
});
```

[Русская версия здесь.]:../ru/custom-settings.md "Пользовательские настройки"
[color scheme]:./color-scheme.md

docs/ru/custom-settings.md
# Пользовательские настройки

**[The English version is here.]**

Список всех параметров, которые могут быть указаны при создании экземпляра утилиты.

* [horizontalPosition](#horizontalPosition)
* [verticalPosition](#verticalPosition)
* [maxWidth](#maxWidth)
* [maxHeight](#maxHeight)
* [fontSize](#fontSize)
* [showClearTitle](#showClearTitle)
* [showOutputDirection](#showOutputDirection)
* [allowColorization](#allowColorization)
* [consoleColorScheme](#consoleColorScheme)
* [popupColorScheme](#popupColorScheme)
* [serverColorScheme](#serverColorScheme)
* [consoleEOL](#consoleEOL)
* [tabChar](#tabChar)
* [decimalPlaces](#decimalPlaces)
* [run](#run)

Параметры должны передаваться через объект, например:
```javascript
const UduJS = require('udujs/Client');
const Debug = new UduJS({
  maxWidth: 'auto',
  fontSize: '1.2em',
  decimalPlaces: 4,
  popupColorScheme: 'dark',
});
```

<a name="horizontalPosition"></a>
## horizontalPosition <code>строка | только клиент</code>

Горизонтальное положение всплывающего сообщения в окне браузера.
Допустимые значения: <code>left</code> или <code>right</code>.
По умолчанию установлено значение <code>left</code>.
```javascript
const Debug = new UduJS({
  horizontalPosition: 'right',
});
```

<a name="verticalPosition"></a>
## verticalPosition <code>строка | только клиент</code>

Вертикальное положение всплывающего сообщения в окне браузера.
Допустимые значения: <code>top</code> или <code>bottom</code>.
По умолчанию установлено значение <code>bottom</code>.
```javascript
const Debug = new UduJS({
  verticalPosition: 'top',
});
```

<a name="maxWidth"></a>
## maxWidth <code>число, строка | только клиент</code>

Максимальная ширина всплывающего сообщения.
Допустимые значения: <code>number</code> или <code>auto</code>.
По умолчанию установлено значение <code>500</code>.
```javascript
const Debug = new UduJS({
  maxWidth: 'auto',
});
```

<a name="maxHeight"></a>
## maxHeight <code>число | только клиент</code>

Максимальная высота всплывающего сообщения.
Допустимое значение: <code>number</code>.
По умолчанию установлено значение <code>600</code>.
```javascript
const Debug = new UduJS({
  maxHeight: 400,
});
```

<a name="fontSize"></a>
## fontSize <code>строка | только клиент</code>

Размер шрифта во всплывающем сообщении.
Допустимое значение: <code>string</code>.
По умолчанию установлено значение <code>1em</code>.
```javascript
const Debug = new UduJS({
  fontSize: '2em',
});
```

<a name="showClearTitle"></a>
## showClearTitle <code>логический | только клиент</code>

Всплывающая подсказка "Нажмите, чтобы очистить". Если она раздражает, то вы можете отключить её.
Допустимые значения: <code>true</code> или <code>false</code>.
По умолчанию установлено значение <code>true</code>.
```javascript
const Debug = new UduJS({
  showClearTitle: false,
});
```

<a name="showOutputDirection"></a>
## showOutputDirection <code>строка | только клиент</code>

Направление вывода для метода <code>[show()](./client-api.md#show)</code>.
Допустимые значения: <code>window</code> или <code>console</code>.
По умолчанию установлено значение <code>window</code>.
```javascript
const Debug = new UduJS({
  showOutputDirection: 'console',
});
```

<a name="allowColorization"></a>
## allowColorization <code>логический | только клиент</code>

Разрешить окраску отображаемой информации.
Этот параметр применим **только к консоли в браузере**.
Выключите его, если браузер неправильно окрашивает текст в консоли.
Допустимые значения: <code>true</code> или <code>false</code>.
По умолчанию установлено значение <code>true</code>.
```javascript
const Debug = new UduJS({
  allowColorization: false,
});
```

<a name="consoleColorScheme"></a>
## consoleColorScheme <code>строка | только клиент</code>

[Цветовая схема] для консоли в браузере.
Допустимые значения: <code>dark</code>, <code>bright</code> или <code>custom</code>.
По умолчанию установлено значение <code>dark</code>.
```javascript
const Debug = new UduJS({
  consoleColorScheme: 'bright',
});
```

<a name="popupColorScheme"></a>
## popupColorScheme <code>строка | только клиент</code>

[Цветовая схема] для всплывающего сообщения в браузере.
Допустимые значения: <code>dark</code>, <code>bright</code> или <code>custom</code>.
По умолчанию установлено значение <code>bright</code>.
```javascript
const Debug = new UduJS({
  popupColorScheme: 'dark',
});
```

<a name="serverColorScheme"></a>
## serverColorScheme <code>строка | только сервер</code>

[Цветовая схема] для консоли на сервере.
Допустимые значения: <code>dark</code>, <code>bright</code> или <code>custom</code>.
По умолчанию установлено значение <code>dark</code>.
```javascript
const Debug = new UduJS({
  serverColorScheme: 'custom',
});
```

<a name="consoleEOL"></a>
## consoleEOL <code>строка</code>

Символ новой строки для сообщений в консоли.
Рекомендуемые значения: <code>\r\n</code> - Windows-стиль, <code>\n</code> - Linux-стиль.
Допустимое значение: <code>string</code>.
По умолчанию установлено значение <code>\n</code>.
```javascript
const Debug = new UduJS({
  consoleEOL: '\r\n',
});
```

<a name="tabChar"></a>
## tabChar <code>строка</code>

Размер одного символа табуляции для форматирования сообщений.
Шаг отступа для вложенных структур данных.
Допустимое значение: <code>string</code>.
По умолчанию указано два символа пробела.
```javascript
const Debug = new UduJS({
  tabChar: '..',
});
```

<a name="decimalPlaces"></a>
## decimalPlaces <code>число</code>

Число десятичных знаков для отображения времени выполнения в RTT-методах (точность измерения).
Укажите 0, чтобы отключить десятичную часть.
Допустимое значение: <code>number</code>.
По умолчанию установлено значение <code>2</code>.
```javascript
const Debug = new UduJS({
  decimalPlaces: 4,
});
```

<a name="run"></a>
## run <code>логический</code>

Этот параметр позволяет быстро отключить все компоненты утилиты.
Допустимые значения: <code>true</code> или <code>false</code>.
По умолчанию установлено значение <code>true</code>.
```javascript
const Debug = new UduJS({
  run: false,
});
```

[The English version is here.]:../en/custom-settings.md "Custom settings"
[Цветовая схема]:./color-scheme.md

CHANGELOG.md
# Change Log

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

<a name="1.0.4"></a>
## 1.0.4 (2017-12-18)


### Bug Fixes

* **lib:** updated version number in the NPM package



<a name="1.0.3"></a>
## 1.0.3 (2017-12-18)


### Bug Fixes

* **structure files:** changed the structure of the location of source files ([424705b](https://github.com/Heliax44/udujs/commit/424705b))



<a name="1.0.2"></a>
## 1.0.2 (2017-12-18)


### Bug Fixes

* **src:** adding entry points for direct access to the "Client" and "Server" modules ([ae964e3](https://github.com/Heliax44/udujs/commit/ae964e3))



<a name="1.0.1"></a>
## 1.0.1 (2017-12-17)

* Initial release.

lib/config.js
module.exports = {
  popupMsg: {
    fontSize: '1em', // The font size in the pop-up message.
    showClearTitle: true, // The "Click to clear" tooltip on the pop-up message.
    horizontalPosition: 'left', // Horizontal position of pop-up message: "left" or "right".
    verticalPosition: 'bottom', // Vertical position of pop-up message: "top" or "bottom".
    maxWidth: 500, // Maximum width of pop-up message.
    maxHeight: 600, // Maximum height of pop-up message.
    // TODO: Implement customizing the direction of adding new messages: from above or below.
    messageDirection: 'above', // "above" or "below".
    bottomIndent: 22, // An indent at the bottom of the browser for the status line to be visible.
  },
  serviceApp: {
    showOutputDirection: 'window', // Output direction for the "show()" method: "window" or "console".
    consoleEOL: '\n', // A newline character for messages in the console. Recommended values: "\r\n" - Windows-style, "\n" - Linux-style.
    tabChar: '  ', // The size of one tab character for formatting messages.
    appName: 'UduJS',
    appDescription: 'A simple universal debugging utility for JavaScript code.',
    appVersion: '1.0.4',
    consoleColorScheme: 'dark', // The default color scheme for the console in the browser. "dark", "bright" or "custom".
    popupColorScheme: 'bright', // The default color scheme for a pop-up message in the browser. "dark", "bright" or "custom".
    serverColorScheme: 'dark', // The default color scheme for the console on the server. "dark", "bright" or "custom".
    allowColorization: true, // Allow coloring of the displayed information.
  },
  performance: {
    decimalPlaces: 2, // The number of decimal places to display the execution time in RTT-methods.
  },
  runtime: {
    run: true, // Allow the utility to run, or not.
    globalIndentSize: 0, // The current value of the indent in the output stream.
    msgContainer: null, // Container for pop-up message: does not exist by default.
    observer: null, // Container for pop-up observer: does not exist by default.
    colorScheme: {}, // A color scheme that will be loaded based on preferences.
    cache: {},
  },
};
